frontend/src/modules/_shared/SensitivityProcessing/sensitivityProcessing.ts

```diff
@@ -0,0 +1,66 @@
+import type { EnsembleSensitivities } from "@framework/EnsembleSensitivities";
+import { SensitivityType } from "@framework/EnsembleSensitivities";
+
+import {
+    computeReferenceAverage,
+    filterSensitivityResponses,
+    sortSensitivityResponses,
+    validateReferenceSensitivity,
+} from "./helpers";
+import { processMonteCarloSensitivity } from "./processMontecarlo";
+import { processScenarioSensitivity } from "./processScenario";
+import { type EnsembleScalarResponse, type SensitivityResponseDataset, type SensitivitySortOrder } from "./types";
+
+// Domain specific thing (Reference realization?). This case name should be ignored.
+const IGNORED_CASE = "ref";
+
+function processSensitivities(
+    sensitivities: EnsembleSensitivities,
+    ensembleResponse: EnsembleScalarResponse,
+    referenceAverage: number,
+) {
+    return sensitivities
+        .getSensitivityArr()
+        .filter((s) => s.name !== IGNORED_CASE)
+        .map((sensitivity) => {
+            switch (sensitivity.type) {
+                case SensitivityType.SCENARIO:
+                    return processScenarioSensitivity(sensitivity, ensembleResponse, referenceAverage);
+                case SensitivityType.MONTECARLO:
+                    return processMonteCarloSensitivity(sensitivity, ensembleResponse, referenceAverage);
+                default:
+                    throw new Error(`Sensitivity type ${sensitivity.type} not supported`);
+            }
+        });
+}
+
+export const computeSensitivitiesForResponse = (
+    sensitivities: EnsembleSensitivities,
+    ensembleResponse: EnsembleScalarResponse,
+    referenceSensitivity: string,
+    barSortOrder: SensitivitySortOrder,
```

Comment:
This utility is used for both TornadoChart and table, thus the `SortBy` (if renamed) yields for more than bars in the TornadoChart